lib/node/runner/isaac_state_manager.go

```diff
@@ -43,26 +44,46 @@ func NewISAACStateManager(nr *NodeRunner, conf common.Config) *ISAACStateManager
 	}
 
 	genesisBlock := block.GetGenesis(nr.storage)
-	p.genesis = genesisBlock.Header.Timestamp
+	p.firstConsensusBlockTime = genesisBlock.Header.Timestamp // temporarily initialized with genesis
+	p.isSetFirstConsensusBlockTime = false
```

Comment:
`p.isSetFirstConsensusBlockTime` is for checking the time of first confirmed block, right? If right, I think there is more simple way without `p.isSetFirstConsensusBlockTime`

```go
p.firstConsensusBlockTime = time.Time{}
return p
}

func (sm *ISAACStateManager) setFirstConsensusBlockTime() {
  if !p.firstConsensusBlockTime.IsZero() {
    return
 }

  b := sm.nr.Consensus().LatestBlock()
  if b.Height == common.GenesisBlockHeight {
  	return
  }
  
  blk, err := block.GetBlockByHeight(sm.nr.Storage(), common.FirstConsensusBlockHeight)
  if err != nil {
  	return
  }
  sm.firstConsensusBlockTime = blk.Header.Timestamp
  sm.nr.Log().Debug("set first consnsus block time", "time", sm.firstConsensusBlockTime)
}
```